src/main/kotlin/com/terraformation/backend/accelerator/db/VoteStore.kt

```diff
@@ -101,4 +120,50 @@ class VoteStore(
         .where(PROJECTS.ID.eq(projectId))
         .fetchOne(COHORTS.PHASE_ID) ?: throw ProjectNotInCohortException(projectId)
   }
+
+  private fun updateProjectVoteDecisions(
+      projectId: ProjectId,
+      phase: CohortPhase,
+      now: Instant = clock.instant()
+  ): VoteDecisionModel {
+    val votes =
+        dslContext
+            .select(PROJECT_VOTES.VOTE_OPTION_ID)
+            .from(PROJECT_VOTES)
+            .where(PROJECT_VOTES.PROJECT_ID.eq(projectId))
+            .and((PROJECT_VOTES.PHASE_ID.eq(phase)))
+            .fetch { it.value1() }
+
+    val decision =
+        if (votes.isEmpty()) {
+          // If no voter exists (e.g. last vote deleted)
+          null
+        } else if (votes.any { it == null }) {
+          // If one or more voter has not submitted a vote
+          null
+        } else {
+          val counts = votes.groupingBy { it!! }.eachCount()
+          val highestCount = counts.maxOf { it.value }
+          val majority = counts.filterValues { it == highestCount }.keys
+
+          // Return relative majority only if not a tie
+          if (majority.size == 1) majority.first() else null
```

Comment:
It occurred to me that this could be `majority.singleOrNull()` but I think they way you have it is probably a bit better because it makes the intent more explicit.